PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetPP.cxx

```diff
@@ -672,19 +680,55 @@ Double_t AliAnalysisTaskJetPP::EstimateLocalBg(AliJetContainer *jetCont,AliParti
 
       trackpt = track->Pt();
       tracketa = track->Eta();
-      trackphi = track->Phi()-TMath::Pi();
+      trackphi = Convert(track->Phi());
       X=tracketa-highjeteta;
       Y=Convert(trackphi-perpphi1);
       Z=Convert(trackphi-perpphi2);     
 
-      if(X*X+Y*Y<coner2 || X*X+Y*Y<coner2){
+      if((X*X+Y*Y) < coner2 || (X*X+Z*Z) < coner2){
          bgpt=bgpt+trackpt;
       }
    }
-   rhoCone = bgpt/(2*TMath::Pi()*coner2); //take the median of pTjet/Ajet
+   rhoCone = bgpt/(2*TMath::Pi()*coner2); //calculate the pT density
     
    return rhoCone; 
 }
+//_______________________________________________________________________
+Double_t AliAnalysisTaskJetPP::EstimateBgKTCMS(AliJetContainer *jetCont){
+   //Estimate rho by the improved CMS method from kT jets
+   Double_t rhoCMS    = 0.0;
+ 
+   if(!jetCont) return rhoCMS;   
+
+   AliEmcalJet*  jet        = NULL;
+   static Double_t rhovec[999];
+   Int_t nJetAcc = 0;
+   Double_t jetpt = 0.0;
+   Double_t physJetArea = 0.0;
+   Double_t allJetArea = 0.0; 
+
+   jetCont->ResetCurrentID();
```

Comment:
These iterators are deprecated. Their use can lead to unexpected behaviour. Please use the C++ standard iterators as explained in the [documentation](http://alidoc.cern.ch/AliPhysics/master/_r_e_a_d_m_econtainers.html#emcalContainerIterateTechniques)